pkg/dash/dash.go

```diff
@@ -129,39 +125,42 @@ func NewRunner(ctx context.Context, logger log.Logger, options Options) (*Runner
 	return &r, nil
 }
 
-func (r *Runner) Start(ctx context.Context, logger log.Logger, options Options, startupCh, shutdownCh chan bool) error {
-	kubeConfigPath := ocontext.KubeConfigChFrom(ctx)
+func (r *Runner) Start(options Options, startupCh, shutdownCh chan bool) error {
+	logger := internalLog.From(r.ctx)
 	go func() {
-		if err := r.dash.Run(ctx, startupCh); err != nil {
+		if err := r.dash.Run(r.ctx, startupCh); err != nil {
 			logger.Debugf("running dashboard service: %v", err)
 		}
-		return
 	}()
 
 	if !r.apiCreated {
-		go func() {
-			if r.dash != nil {
-				var err error
-				if options.KubeConfig, err = ValidateKubeConfig(logger, options.KubeConfig, r.fs); err != nil {
-					logger.Infof("waiting for kube config ...")
-					options.KubeConfig = <-kubeConfigPath
-				}
-
-				if options.KubeConfig != "" {
-					logger.Debugf("Loading configuration: %v", options.KubeConfig)
-					r.startAPIService(ctx, logger, options)
-					return
-				} else {
-					logger.Errorf("unexpected empty kube config")
-					return
-				}
-			}
-		}()
-	} else {
-		r.startAPIService(ctx, logger, options)
+		logger.Infof("waiting for kube config ...")
+		options.KubeConfig = <-ocontext.KubeConfigChFrom(r.ctx)
```

Comment:
This blocks which is fine because it needs to wait if the user adds a kubeconfig, however it will hang on sigint/sigterm without shutting down plugins. In other words, octant will hang if the users tries to exit in the loading api state via `ctrl + c` without a way to recover

Reply:
oh that makes sense, thank you! I'll fix this up.